Improve theme styles

Gustl22 opened the pull request:
Font-families added: `thin`, `light`, `medium`, `black`, `condensed`
If lower API do not support these fonts, they're automatically replaced.
Themes:
- use same font for building names and amenities, shops etc. cause building names describe amenities (changed in default and newtron, other were already right); was kind of peculiar that the same name was written in two fonts...font is called `caption-small-blue` in most cases.
- changed railway color to a more unobtrusive color (so it's no more kind of a warning)
- changed amenity color (`caption-small-blue`) to a more unobtrusive color and use `medium` family/style, cause in big cities too much and big labels are displayed, in my opinion.
- changed house-number style/family to `condensed`, so more space is available
  
BTW a short question: Is there zoom-level 18 available for themes?

devemux86 commented:
Seems nice!

Note: render / graphics changes in the core should be implemented for all supported platforms.
Certainly we all have desktop to test, which now isn't working.
Other platforms should have fallback solutions for not breaking user experience.

Gustl22 commented:
> Certainly we all have desktop to test, which now isn't working.
Oh sorry you're right! Forgot about that. Can try to fix it!

devemux86 commented:
> Is there zoom-level 18 available for themes?

If you mean Mapsforge maps, the `MapFileTileSource` has default max extraction zoom 17. Can change it in apps, but should be careful for cases like #231.
Better way is implement #393 which would help all vector map sources, which normally end data retrieval at zoom 14-15.

devemux86 commented:
> But unfortunately I didn't work with iOS yet.

@Longri can help here! :slightly_smiling_face:

devemux86 commented:
> If lower API do not support these fonts, they're automatically replaced.

Probably you mean Android, can you clarify that, like the docs?
Or mean via support library, which shouldn't be considered mandatory.

Gustl22 commented:
Thanks emux, for your answers :)

Now added an implementation for **AWT** and iOS, but the **iOS** is NOT tested!!
Thanks @Longri, simply do, when you want. My biggest fear is that the Font loaded for every label separately.. don't know.

Awt: unfortunately awt not adapt all weight steps (e.g. medium and regular seem pretty the same). But shouldn't be a problem.

> Probably you mean Android, can you clarify that, like the docs?

Yes meant [Android itself](https://developer.android.com/reference/android/graphics/Typeface.html#create(java.lang.String,%20int)), but now all implementations should replace/interpret all font types.

devemux86 commented:
@Gustl22 
Thanks for the changes!
- Doesn't the `GwtPaint` also need adjustments?
- Or similar theme changes also in `openmaptiles.xml`?

@Longri 
Please do, it's important to not break any platform. :slightly_smiling_face:

Gustl22 commented:
> Doesn't the GwtPaint also need adjustments?

OK :) I'll check that tomorrow.
> Or similar theme changes also in openmaptiles.xml?

Didn't went through all themes, but can fix this of course.

devemux86 commented on vtm-desktop/src/org/oscim/awt/AwtPaint.java:
```diff
@@ -169,9 +185,10 @@ public void setTextSize(float textSize) {
     @Override
     public void setTypeface(FontFamily fontFamily, FontStyle fontStyle) {
-        this.fontName = getFontName(fontFamily);
+        Map<Attribute, Object> tmpAttributes = new HashMap<>(this.TEXT_ATTRIBUTES);
```
Better avoid here the object creation for each rule or label, can think a way to somehow improve that.

Gustl22 replied:
Moved it in one method, as methods cannot return two objects, and a new class would cause new instances. Thought of an final static instance, too, but has to be synchronized and reset and refilled. So this is the better alternative I think. If you have improvements, can edit of course :)

devemux86 replied:
Thanks, that seems better!

Gustl22 commented:
Added a new commit. Now hopefully all platforms are supported (vague iOS).  
Adjusted the themes files. But I cannot guarantee if I haven't missed something. E.g. Openmaptiles do not support buildings and higher zoom-levels yet. In general all themes need some unification in the future. Not as easy done as said :)

devemux86 commented:
@Gustl22 when you're ready, I can review the changes and merge them in a separate branch for @Longri to see the iOS too.

Gustl22 commented:
Yes u can review them, thanks!

Longri commented:
I have now checked out the master and merged the PR locally.

Then I took all *. gradle files from an older version and copied them into the branch.

I can confirm that the changed code, from PR, works on iOS-Device.

But I had to make some changes to the native SVG code. Here is still native code from the 32-bit (x86) which is no longer supported.

I will create a separate PR containing my changes without including the *. gradle files.

devemux86 commented:
Thanks @Longri for the testing!

I'll review also the changes and merge later.

Longri commented:
if you create a VTM-SNAPSHOT after the merge process and load it to Maven, then I will switch to this SNAPSHOT in my project and can easily see if the library creation is correct.

devemux86 commented:
Every successful `master` build is automatically published on [Sonatype OSS Repository Hosting](https://oss.sonatype.org/content/repositories/snapshots/org/mapsforge/) and can be used via [master-SNAPSHOT](https://github.com/mapsforge/vtm/blob/master/docs/Integration.md#snapshots) version.

Longri commented:
BUT wait for my PR for the native! 

I'll try to get it done by this afternoon!
(Family wakes up, breakfast time)

devemux86 commented:
Do you mean I merge in a branch where you can post a pull request?
In that case you'll have to pull the branch locally to test.

Longri commented:
Nope, my PR is based on master! the changes are only the native- SVGh (for iOS)

devemux86 commented:
I didn't understand the "wait", if it's not related to this PR, you can post it whenever you can(?).

Longri commented:
The' wait' was related to the SNAPSHOT, but you just made that clear! Sorry

devemux86 commented:
Thanks for the very nice improvements and checking all platforms!

Squashed and merged via 067f08834d0e7eefca0ff59f08e7ea5df458ef79.
